src/Umbraco.Web.UI.Client/src/controllers/navigation.controller.js

```diff
@@ -391,6 +399,8 @@ function NavigationController($scope, $rootScope, $location, $log, $q, $routePar
     $scope.selectLanguage = function (language) {
 
         $location.search("mculture", language.culture);
+        // add the selected culture to a cookie so the user will log back into the same culture later on
+        $cookies.put("UMB_MCULTURE", language.culture, {path: "/", expires: "2100-01-01"});
```

Comment:
Could you use `max-age` instead and set it to 1 year? This is a tiny concern but some browser seems to be affected by the 2038 bug. Also, what if we're still using this site in the year 2101? 😉

Reply:
Absolutely, I'll have a look at it 👍

Reply:
@nul800sebastiaan done. Unfortunately $cookies doesn't support `max-age` (see https://docs.angularjs.org/api/ngCookies/provider/$cookiesProvider#defaults), but it works just as fine setting the cookie the good ol' JS way (`document.cookie = ...`).